gulpfile.js

```diff
@@ -0,0 +1,30 @@
+const gulp = require('gulp');
+const gulpif = require('gulp-if');
+const typescript = require('gulp-typescript');
+const mergeStream = require('merge-stream');
+const polymer = require('polymer-build');
+const Transform = require('stream').Transform;
+
+const PolymerProject = polymer.PolymerProject;
+
+let polymerProject = new PolymerProject({
+  entrypoint: 'index.html',
+  shell: 'src/my-app.html',
+});
+
+let tsProject = typescript.createProject('tsconfig.json');
+
+gulp.task('build', ['build:ts'], () => {
+  return mergeStream(
+      // tsProject.src().pipe(typescript(tsProject)),
+      polymerProject.sources(),
+      polymerProject.dependencies()
+    )
+    .pipe(gulp.dest('build'));
+});
+
+gulp.task('build:ts', () => {
+  return tsProject.src()
```

Comment:
gulp-typescript really only seems to behave when you use it's `.src()` mechanism, yet their docs suggest that `gulp.src()` is better for integration with watch. Not sure what the best pattern here is, but it's not integrating with `PolymerProject.sources()` particularly well.